shopinvader/models/shopinvader_backend.py

```diff
@@ -434,3 +434,11 @@ def write(self, values):
             self._get_id_from_auth_api_key.clear_cache(self.env[self._name])
         with self._keep_binding_sync_with_langs():
             return super(ShopinvaderBackend, self).write(values)
+
+    def _get_cart_pricelist(self, partner):
+        """Retrieve pricelist to be used for the cart.
+
+        Hook here to customize the pricelist used for given customer on cart.
+        """
+        # TODO: shouldn't we return partner.property_product_pricelist if valued?
```

Comment:
I'm not the one who played with pricelists. Maybe @Cedric-Pigeon ?

Reply:
hum sorry, do not remember clearly. I guess we should test if self.pricelist_id is null and use the customer in one in that case

Reply:
@sebastienbeau any feedback on this?

Reply:
I probably found the answer: the same pricelist is used everywhere to have always the same price for each customer.

ATM you cannot have prices per customer in base Shopinvader because otherwise you won't have the same prices in the product index, leading to different prices in cart and product pages.

Reply:
FTR I'm working on this to solve the issue for product pages https://github.com/shopinvader/odoo-shopinvader/pull/803